### DISCUSSION TITLE
Setting custom colours on progress component

### DISCUSSION THREAD

**Discussion by hades200082** [saadeghi/daisyui#1594] [Ideas / feature requests]
Currently, only the colours defined in the DaisyUI theme may be used for the progress component. 

I'd like to be able to apply different colours from Tailwind also.

Use-case: I want to create a bar that has different colours depending on the value.

```jsx
<progress className={cn(
      "progress",
      {
          "progress-green-500": score >= 8,
          "progress-lime-400": score < 8 && score  >= 5,
          "progress-yellow-400": score  < 5 && score  >= 4,
          "progress-orange-500": score  < 4 && score  >= 3,
          "progress-red-600": score  < 3,
      }
  )} value={score * 10} max="100"></progress>
```

**Comment by saadeghi** [comment]
You need to add 2 class names (one for Webkit and one for Firefox):
```
[&::-webkit-progress-value]:bg-red-500
[&::-moz-progress-bar]:bg-red-500
```
Example: https://play.tailwindcss.com/xx1ROaGl1Z

**Reply by hades200082** [reply]
Thanks for the workaround. 👍 

Would still be nice to see the native Tailwind colours supported natively in the component.

**Reply by saadeghi** [reply]
@hades200082 It is possible to do so but there would be problems with that idea: 

1. using Tailwind CSS color names (like `red-500` etc.) on daisyUI components breaks the theme functionality of daisyUI. All daisyUI color names have dynamic values (coming from a CSS variable) and if we use a constant color name for the component it can never adapt the current theme.  
I would suggest to try using daisyUI color names. You won't regret it and it gives you much more control during development and helps you commit to your design system. Read more: https://daisyui.com/docs/colors/

2. daisyUI components usually use multiple color values underneath. For example, `btn-primary` is using `primary` color for the background, `primary-focus` for the hover state and `primary-content` for the text. If we were able to use `btn-red-500` how can we know about the hover color and text color? So if we were able to use any Tailwind CSS color for a daisyUI button we should've set 3 colors for it something like this: (pseudo code) `btn-red-500 hover:btn-red-600 btn-text-red-100`... and that wouldn't help at all because currently we can already do that like this `bg-red-500 hover:bg-red-600 text-red-100`

3. If we could generate all daisyUI components with all Tailwind CSS color name, it would quickly result a lot of inaccessible color contrasts.

**Reply by hades200082** [reply]
@saadeghi Thanks for the detailed response and for the most part o agree. 

However I have a use case that is very specific to progress bars (and the round ones too). 

I need to color the bars based on the value. For example, I need lower values to be closer to red and higher values to be closer to green with the middle of the range passing through yellow or amber. 

This adds to ux by allowing users to easily differentiate value ranges at a glance. 

It is very common to present data like this.

**Reply by saadeghi** [reply]
I understand. And to be fair on most components you can change the color using one Tailwind CSS utility class. But for progress, we should blame the browsers because at this time, Firefox and WebKit don't support a unified CSS rule for coloring it. In result, Tailwind doesn't offer a utility class for it because it's a prefixed browser specific CSS rule. So there's no way unless we set both `-webkit-progress-value` and `-moz-progress-bar` so we get the same result on all browsers.
